test/functional/test_framework/authproxy.py

```diff
@@ -78,7 +78,9 @@ def __init__(self, service_url, service_name=None, timeout=HTTP_TIMEOUT, connect
         passwd = None if self.__url.password is None else self.__url.password.encode('utf8')
         authpair = user + b':' + passwd
         self.__auth_header = b'Basic ' + base64.b64encode(authpair)
-        self.timeout = timeout
+        # larger values of the timeout cause an "Invalid argument"
+        # socket exception in Python's HTTP client library, so clamp it
```

Comment:
nit: Just to clarify what's being clamped, though the code is self-explanatory.

```suggestion
        # socket exception in Python's HTTP client library
        # to avoid this, we clamp the timeout value
```

Or something similar.